lib/handler.ex

```diff
@@ -26,23 +26,36 @@ defmodule Handler do
 
   """
   def run(fun, opts \\ []) do
-    # default options to 20min and 1GB of RAM
-    max_ms = Keyword.get(opts, :max_ms, 1_200_000)
-    max_heap_bytes = Keyword.get(opts, :max_heap_bytes, 1024 * 1024 * 1024)
-    max_heap_words = div(max_heap_bytes, :erlang.system_info(:wordsize))
+    max_ms = max_ms(opts)
+    max_heap_bytes = max_heap_bytes(opts)
 
     old_trap_exit = Process.flag(:trap_exit, true)
 
     result =
       fun
-      |> kickoff_fun(max_heap_words)
+      |> kickoff_fun(max_heap_bytes)
       |> await_results(max_ms, max_heap_bytes)
 
     Process.flag(:trap_exit, old_trap_exit)
 
     result
   end
 
+  @doc false
+  def bytes_to_words(max_bytes) when is_integer(max_bytes) do
+    div(max_bytes, :erlang.system_info(:wordsize))
+  end
+
+  @doc false
+  def max_heap_bytes(opts) do
+    Keyword.get(opts, :max_heap_bytes, 10 * 1024 * 1024)
+  end
+
+  @doc false
+  def max_ms(opts) do
+    Keyword.get(opts, :max_ms, 60_000)
+  end
```

Comment:
I broke them out into functions so that I could call them from the pool code without having to duplicate the code. Totally open to other ways of making sure we apply the same defaults in the pool and in this code